python/tank/platform/application.py

```diff
@@ -117,7 +120,40 @@ def engine(self):
         The engine that this app is connected to.
         """
         return self.__engine
-        
+
+    @property
```

Comment:
Would it make sense of apps and frameworks to share a common base class where the logging logic could be?  Seems like there is a lot of really similar code between those two bundle types for this.